File: Basic Sorting Algorithms/BubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

void bubble_sort(vector<int> & v) {
    int n = v.size();
    for(int i=0; i<n-1; i++) {
        for(int j=0; j<n-1-i; j++) {
            if(v[j] > v[j+1]) {
                swap(v[j], v[j+1]);
            }
        }
    }
}

int main() {
    vector<int> v = {7,6,5,4,3,2,1};
    bubble_sort(v);
    for(int num : v) {
        cout<<num<<" ";
    }
}

File: Basic Sorting Algorithms/BubbleSort.js
function bubble_sort(arr) {
  let n = arr.length;
  for (let times = 1; times <= n - 1; times++) {
    for (let j = 0; j <= n - times - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
      }
    }
  }

  for (let num of arr) {
    console.log(num);
  }
}

let arr = [-2, 3, 4, -1, 5, -12, 6, 1, 3];
bubble_sort(arr);
